From 53491bd3124d3da3373e8c9ff9ce5b8ab12ea1f0 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Mon, 25 Jan 2021 20:07:13 +0800
Subject: [PATCH] drm/rockchip: vop2: alpha_cal_mode use ALPHA_NO_SATURATION is
 better

set Ad0" = Ad0_" to instead of Ad0" = Ad0_" + (Ad0_" >> 7), than VOP
some alpha result is same with GPU output.

Change-Id: If05f561c16b4eb6e46671f4a130fae0fe50191c2
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index e2dc55d92c73..c709d1b34a79 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -3879,7 +3879,7 @@ static void vop2_parse_alpha(struct vop2_alpha_config *alpha_config,
 		alpha->dst_alpha_ctrl.bits.blend_mode = ALPHA_PER_PIX;
 	else
 		alpha->dst_alpha_ctrl.bits.blend_mode = ALPHA_PER_PIX_GLOBAL;
-	alpha->dst_alpha_ctrl.bits.alpha_cal_mode = ALPHA_SATURATION;
+	alpha->dst_alpha_ctrl.bits.alpha_cal_mode = ALPHA_NO_SATURATION;
 	alpha->dst_alpha_ctrl.bits.factor_mode = ALPHA_SRC_INVERSE;
 }
 
-- 
2.35.3

